benchmarks/solve_ivp_bench.py

```diff
@@ -0,0 +1,480 @@
+
+    Returns:
+        pl.DataFrame: A dataframe containing the timing and error results.
+    """
+    match algo_args["method"]:
+        case Algo.PYRICCATICPP:
+            info = ric.Init(*algo_args["method_args"]["init_args"])
+            solver_args = algo_args["method_args"]["solver_args"]
+            algo_args["method_args"]["init_step_args"] = (info,
+                                                          solver_args["xi"],
+                                                          solver_args["xf"],
+                                                          solver_args["epsilon_h"])
+            init_step = ric.choose_nonosc_stepsize(*algo_args["method_args"]["init_step_args"])
+            solver_args["init_stepsize"] = init_step
+            runtime = timeit.timeit(lambda: ric.evolve(info=info, **solver_args), number=N) / N
+            _, ys, _, _, _, _, _, _, _ = ric.evolve(info=info, **solver_args)
+            ys = np.array(ys)
+            # Compute statistics
+            yerr = np.abs((problem_info.y1 - ys[-1]) / problem_info.y1)
+            print_args = algo_args["method_args"]["print_args"]
+            timing_df = pl.DataFrame({"eq_name":algo_args["function_name"],
+                                      "method": str(algo_args["method"]),
+                                      "eps": algo_args["eps"],
+                                      "relerr": yerr,
+                                      "walltime": runtime,
+                                      "errlessref": bool(yerr < problem_info.relative_error),
+                                      "problem_params": problem_info.print_params(),
+                                      "params": f"n={print_args['n']};p={print_args['n']};epsh={print_args['epsh']}"})
+            return timing_df
+        # All Python IVPs use the same scheme
+        case _:
+            runtime = timeit.timeit(lambda: solve_ivp(**algo_args["method_args"]), number=N) / N
+            sol = solve_ivp(**algo_args["method_args"])
+            yerr = np.abs((sol.y[0, -1] - problem_info.y1) / problem_info.y1)
+            timing_df = pl.DataFrame({"eq_name":algo_args["function_name"],
+                                      "method": str(algo_args["method"]),
+                                      "eps": algo_args["eps"],
+                                      "relerr": yerr,
+                                      "walltime": runtime,
+                                      "errlessref": bool(yerr < problem_info.relative_error),
+                                      "problem_params": problem_info.print_params(),
+                                      "params": f"rtol={algo_args['method_args']['rtol']};atol={algo_args['method_args']['atol']}"})
+            return timing_df
+
+timing_dfs = []
+for problem_key, problem_item in problem_dictionary.items():
+    for algo, algo_params in algorithm_dict.items():
+        for algo_iter in list(itertools.product(*algo_params["args"])):
```

Comment:
@fruzsinaagocs does it make sense here to do all combinations of the tolerances? Or do we want to be more careful with the sets of tolerances we test over?